packages/console/src/hooks/use-tenant-scope-listener.ts

```diff
@@ -0,0 +1,70 @@
+import { Prompt, useLogto } from '@logto/react';
+import { getTenantOrganizationId } from '@logto/schemas';
+import { useContext, useEffect, useState } from 'react';
+
+import { TenantsContext } from '@/contexts/TenantsProvider';
+import { saveRedirect } from '@/utils/storage';
+
+import useCurrentTenantScopes from './use-current-tenant-scopes';
+import useRedirectUri from './use-redirect-uri';
+
+const useTenantScopeListener = () => {
+  const { currentTenantId, isInitComplete } = useContext(TenantsContext);
+  const { isAuthenticated, clearAccessToken, clearAllTokens, getOrganizationTokenClaims, signIn } =
+    useLogto();
+  const [tokenClaims, setTokenClaims] = useState<string[]>([]);
+  const [isDecodingClaims, setIsDecodingClaims] = useState(false);
+  const redirectUri = useRedirectUri();
+  const { scopes = [], isLoading } = useCurrentTenantScopes();
+
+  useEffect(() => {
+    (async () => {
+      if (!isAuthenticated || !isInitComplete) {
+        return;
+      }
```

Comment:
we can omit this if this hook will only be used in authed routes. add some comments to clarify its purpose.

Reply:
otherwise every we need to add this to too many places.

Reply:
actually you can move this hook to the component/page folder

Reply:
Ideally yes, but we can't guarantee if this hook will be used outside of authed routes one day. So I added this as a precaution.

Reply:
Discussed offline and decided to put this hook in `ConsoleContent` folder and use it privately in `ConsoleContent` page